app/src/main/java/com/trongntd/themovie/api/MovieConnector.java

```diff
@@ -0,0 +1,56 @@
+package com.trongntd.themovie.api;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.trongntd.themovie.entity.Movie;
+import com.trongntd.themovie.entity.MovieList;
+import com.trongntd.themovie.entity.MovieVideo;
+
+import java.util.List;
+
+import retrofit2.Callback;
+import retrofit2.Retrofit;
+import retrofit2.converter.gson.GsonConverterFactory;
+
+public class MovieConnector {
```

Comment:
Singleton will cause the app harder to test. Some approachs we can think of here:
1. Abstract factory (recommended).
2. Just create a new instance every time we need it.

We can think about using dependency injection here for a better test by providing an interface for this class. Then later we can simply mock an inteface instead of using this class for test.

Reply:
Yup, I fixed it on branch "home_page". Thanks